### ISSUE TITLE
Translator fails for repeated kernel with different arguments

### ISSUE THREAD

**Issue by spjammy** [OP2/OP2-Common#109]
Hi,

The translator fails for repeated kernels. Example,

op_par_loop(test, "test1", faces,
op_arg_dat( a, -1, OP_ID, 1, OP_READ),
op_arg_dat( a, 0, some_mapping , 1, OP_READ),

**Comment by reguly** [comment]
The kernel specs you wrote down are missing arguments and commas.
op_par_loop(test, "test1", faces,
op_arg_dat( a, -1,OP_ID, 1, "double",OP_READ),
op_arg_dat( c,  0, some_mapping , 1, "double",OP_READ),
op_arg_dat( c, 1, some_mapping , 1, "double",OP_READ));

op_par_loop(test, "test1", faces,
op_arg_dat( d, -1,OP_ID, 1, "double",OP_READ),
op_arg_dat( e, 0, some_mapping , 1, "double",OP_READ),
op_arg_dat( e, 1, some_mapping , 1, "double",OP_READ));

This does work for me with the master branch. Please confirm and close the issue if it works.

**Comment by spjammy** [comment]
Sorry I missed the arguments double while creating the representative case. 

here is the actual code
// Checking internal faces 
    op_par_loop(check_normals, "normals", faces,
                  op_arg_dat(p_face_normal, -1, OP_ID, 2, "double", OP_RW),
                  op_arg_dat(p_cell_centroid, 1, opface_cell_map, 2, "double", OP_READ),
                  op_arg_dat(p_face_centre, -1, OP_ID, 2, "double", OP_READ));
                        //-------CHECKING BOUNDARY FACE NORMAL-----//
    op_par_loop(check_normals, "normals_b", bfaces,
                  op_arg_dat(p_face_normal_b, -1, OP_ID, 2, "double", OP_RW),
                  op_arg_dat(p_cell_centroid, 0, opboundary_cell_map, 2, "double", OP_READ),  
                  op_arg_dat(p_face_centre_b, -1, OP_ID, 2, "double", OP_READ) );
It translates well. But fails in compilation

>>> make solver_genseq

check_normals.h: In function ‘void check_normals(double*, const double*, const double*)’:
check_normals.h:1:13: error: redefinition of ‘void check_normals(double*, const double*, const double*)’
 inline void check_normals(double *face_normals, const double *cell_centroid, const double *face_centre)
             ^
In file included from check_normals_seqkernel.cpp:6:0,
                 from cfdsolver_seqkernels.cpp:22:
check_normals.h:1:13: note: ‘void check_normals(double*, const double*, const double*)’ previously defined here
 inline void check_normals(double *face_normals, const double *cell_centroid, const double *face_centre)
             ^
In file included from cfdsolver_seqkernels.cpp:23:0:
check_normals_seqkernel.cpp: In function ‘void op_par_loop_check_normals(const char*, op_set, op_arg, op_arg, op_arg)’:
check_normals_seqkernel.cpp:9:6: error: redefinition of ‘void op_par_loop_check_normals(const char*, op_set, op_arg, op_arg, op_arg)’
 void op_par_loop_check_normals(char const *name, op_set set

**Comment by reguly** [comment]
Hmm, this is a use case we haven’t come across before. The mappings are just too different for these two to be compatible ü I recommend that you make a copy of the kernel check_normals as a workaround. Will try to come up with something better.

**Comment by spjammy** [comment]
Yes, I did the workaround currently by copying check_normals  for the boundary loop to check_normals_b.
